### DISCUSSION TITLE
About discord.py

### DISCUSSION THREAD

**Discussion by pymice** [Rapptz/discord.py#7848] [Q&A]
I would like to understand about the discord.py comeback, i read the last announcement, but I didn't understand clearly.

-  Danny come back temporarily to allow discord.py to work in the new discord api? 
- What about slashcommands? Where are? Where's the documentation? Why aren't they in the current version?
- Why can't see "discord.ui" documentation on https://discordpy.readthedocs.io/?
- Are you back? Can we re-develop our bots? (Do you intend to continue?)

I'm confused because not seeing the new components clearly, on documentation or on this repo.

**Comment by CnyAnime** [comment]
Hello @pymice,
Danny is back and is currently developing the new version of d.py. Slash commands and discord.ui are in v2.0 which you can find it at https://discordpy.readthedocs.io/en/master/

**Comment by LeoCx1000** [comment]
Yes! Danny explains it all in [this gist](https://gist.github.com/Rapptz/c4324f17a80c94776832430007ad40e6) (also [slash commands docs](https://discordpy.readthedocs.io/en/master/interactions/api.html#discord-app-commands))

**Comment by fuh-Q** [comment]
Yes, Danny's back. Application commands and the UI Kit are in the v2 beta of the library.

But v2 also introduces a lot of breaking changes, so you will have to change some of your original code. There's a list of breaking changes [here](https://discordpy.readthedocs.io/en/master/migrating.html)

**Comment by Rapptz** [comment]
> Danny come back temporarily to allow discord.py to work in the new discord api?

No. v2 already worked with the new API version before I paused development.

> What about slashcommands? Where are? Where's the documentation? Why aren't they in the current version?

They're in `discord.app_commands` in the v2 beta. The documentation currently resides in https://discordpy.readthedocs.io/en/master/ due to mainly legacy but I plan on moving that over to `/latest/` if I can actually remember to get to it. They're not in the current version because v2 isn't done being developed yet so there are still a lot changes to make. These things take time.

> Why can't see "discord.ui" documentation on https://discordpy.readthedocs.io/?

See above.

> Are you back? Can we re-develop our bots? (Do you intend to continue?)

Yes.

**Reply by pymice** [reply]
Considering I'm making a new bot, should I start from v2 or continue on 1.7v until v2 becomes official?

Welcome back and thank you!

**Comment by pymice** [comment]
Thanks for the links and answers, I understand the state of the library better now.
